h/app.py

```diff
@@ -46,6 +47,32 @@ def create_app(global_config, **settings):
     return config.make_wsgi_app()
 
 
+def publish_after_commit_events(request, response):
+    if request.exception is not None:
+        return
+
+    if len(request.after_commit_events) == 0:
```

Comment:
I think this can just be `if request.after_commit_events:`, but do you need to return here at all? Won't the `while` loop in `publish_all()` just do nothing if `after_commit_events` is `[]`?